lib/rubocop/cop/layout/class_structure.rb

```diff
@@ -264,6 +267,22 @@ def dynamic_constant?(node)
             !(expression.method?(:freeze) && expression.receiver&.recursive_basic_literal?)
         end
 
+        def private_constant?(node)
+          return false unless node.casgn_type? && node.namespace.nil?
+          return false unless (parent = node.parent)
+
+          parent.each_child_node(:send) do |child_node|
+            return true if marks_constant_private?(child_node, node.name)
+          end
+          false
+        end
+
+        def marks_constant_private?(node, name)
```

Comment:
What about a method name like this?

```suggestion
        def marked_as_private_constant?(node, name)
```